### PR TITLE
MCS-1451 Updates to segmentation color output metadata

### PR DESCRIPTION
**Pull request by ThomasSchellenbergNextCentury**
Corresponding Python PR: https://github.com/NextCenturyCorporation/MCS/pull/595

See the description in the Python PR.

### CODE REVIEW

File: unity/Assets/ImageSynthesis/ImageSynthesis.cs [hunk 1]
```diff
@@ -305,10 +305,23 @@ public void OnCameraChange()
...
+        }
+        if (isAdjacentToHole) {
+            objTag = "hole wall";
+        }
+        if (hole) {
+            objTag = "hole";
+        }
+        if (lava) {
+            objTag = "lava";
+        }
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/ImageSynthesis/ImageSynthesis.cs:322]: I wanted all holes, hole walls, and lava areas to have the same tag.

File: unity/Assets/Scripts/MCSMain.cs [hunk 2]
```diff
@@ -694,6 +695,7 @@ private bool CheckIfGoalTargetExists()
 
     private void CleanupFloorAfterCloning() {
         this.floor.isStatic = true;
+        this.floor.SetActive(false);
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/Scripts/MCSMain.cs:698]: I didn't see any particular reason why the floor needed to be active in scenes where we create floor sections (but maybe I missed something important?)

File: unity/Assets/ImageSynthesis/ImageSynthesis.cs [hunk 3]
```diff
@@ -361,13 +379,16 @@ public void OnSceneChange()
 			if (so != null || sop != null) 
 			{
 				colorIds [objColor] = objTag;
-				colorIds [classColor] = classTag;
+                // MCS: Intentionally exclude class tags.
+                // colorIds [classColor] = classTag;
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/ImageSynthesis/ImageSynthesis.cs:383]: AI2-THOR has classifications for different objects, but we don't really use these classifications (most of them are set to IgnoreType), so I commented this out.

File: unity/Assets/ImageSynthesis/ImageSynthesis.cs [hunk 4]
```diff
@@ -361,13 +379,16 @@ public void OnSceneChange()
...
 			} 
 
+            /* MCS: Intentionally exclude these objects.
 			else 
 			{
 				colorIds [objColor] = r.gameObject.name;
 			}
+            */
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/ImageSynthesis/ImageSynthesis.cs:391]: This code added a bunch of GameObjects corresponding to the robot model into the dictionary, which we don't want.

**Review by brianpippin — APPROVED** [review]
This looks good to me but would love for @jaudick to chime in on the floor comment.

File: unity/Assets/Scripts/MCSMain.cs (hunk 2 above)
Comment (jaudick) [reply unity/Assets/Scripts/MCSMain.cs:698]: As long as you set it back to active (which you did above) this seems totally fine to me

Comment (jaudick) [reply unity/Assets/Scripts/MCSMain.cs:698]: As long as this is set back to active which you do on line 503 and passes the oracle tests this should be fine

**Review by jaudick — APPROVED** [review]

**Pull request merged**